src/Identity/Core/src/DataProtectorTokenProvider.cs

```diff
@@ -110,38 +125,58 @@ public virtual async Task<bool> ValidateAsync(string purpose, string token, User
                     var expirationTime = creationTime + Options.TokenLifespan;
                     if (expirationTime < DateTimeOffset.UtcNow)
                     {
+                        Logger.LogWarning(0, "ValidateAsync failed: the expiration time is invalid.");
```

Comment:
I know its a pain, but the pattern we are supposed to use for logging going forward is structured logging, see: https://github.com/aspnet/AspNetCore/blob/master/src/Security/Authentication/Core/src/LoggingExtensions.cs

Can you put these new log messages into a new LoggingExtensions class like that?

Reply:
Yes of course. thank. I did not know about it